src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -142,12 +142,20 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <Target Name="_PublishBuildAlternative"
-          Condition="'$(NoBuild)' != 'true'"
-          DependsOnTargets="Build;$(_CorePublishTargets)" />
+          Condition="'$(NoBuild)' != 'true'">
+    <MSBuild
+      Projects="$(MSBuildProjectFile)"
+      Targets="Build;$(_CorePublishTargets)"
+      Properties="_IsPublishing=true" />
```

Comment:
I don't think this is an acceptable approach. This will create a race condition within the build between

1. `Publish` calling `Build` with `_IsPublishing=true`
2. Other projects (or the solution) calling `Build` without that property set.

In addition, the value of `_IsPublishing` will flow to `ProjectReference`s of the current project, causing more build races.